# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [5]:
data[['race','call']].isnull().sum()/len(data)*100

race    0.0
call    0.0
dtype: float64

No missing values for the relevant columns

In [6]:
data['race'].value_counts()

w    2435
b    2435
Name: race, dtype: int64

In [7]:
data['call'].value_counts()

0.0    4478
1.0     392
Name: call, dtype: int64

There were only 392 calls and of those only 157 for black-sounding resumes.

In [8]:
print('No callbacks black-sounding: {}'.format(2435 - sum(data[data.race=='b'].call))) 
print('No callbacks white-sounding: {}'.format(2435 - sum(data[data.race=='w'].call))) 

No callbacks black-sounding: 2278.0
No callbacks white-sounding: 2200.0


### Question 1
***
Since the problem is to compare two populations, we can use the two sample test. Also Central Limit Theorem does apply since n and p are large.

### Question 2
***
$H_0$: white-sounding resume callbacks = black-sounding resume callbacks

$H_a$: white-sounding resume callbacks > black-sounding resume callbacks

### Question 3
***

In [9]:
p1 = sum(data[data['race'] == 'w']['call']) / sum(data['race'] == 'w')
print('p1 (proportion of white-sounding resume callbacks) = {}'.format(p1))

p2 = sum(data[data['race'] == 'b']['call']) / sum(data['race'] == 'b')
print('p2 (proportion of black-sounding resume callbacks) = {}'.format(p2))

p_all = (sum(data[data['race'] == 'w']['call']) + sum(data[data['race'] == 'b']['call'])) / (sum(data['race'] == 'w') + sum(data['race'] == 'b'))
print('p (proportion of all getting call) = {}'.format(p_all))

p1 (proportion of white-sounding resume callbacks) = 0.09650924024640657
p2 (proportion of black-sounding resume callbacks) = 0.06447638603696099
p (proportion of all getting call) = 0.08049281314168377


In [11]:
standard_error = (p_all*(1-p_all)*(1/2435 + 1/2435))**0.5
t = (p1 - p2) / standard_error
margin_of_error = 1.96 * standard_error
CI1 = p1 - p2 - margin_of_error
CI2 = p1 - p2 + margin_of_error
print('margin of error: {}'.format(margin_of_error))
print('confidence interval is: {}, {}'.format(CI1,CI2))
p_val = 2*(1-stats.norm.cdf(t))
print('p-value is: {}'.format(p_val))

margin of error: 0.015281912310894095
confidence interval is: 0.01675094189855149, 0.04731476652033968
p-value is: 3.983886837577444e-05


### Question 4
***
The study shows that discrimination exists in the US labor market. We reject the null hypothesis saying that the number of white-sounding resumes callbacks will be the same amount as the black_sounding resumes. We find out that in reality if you have a white sounding resume your chances of getting a callback are greater. With a 95% confindence, we can say that the proportion of white-sounding resumes getting callbacks will be greater between 0.016 and 0.047.

### QUESTION 5
***
Race and name are just 2 factors of many possible. The analysis does not show that they are the most important. To find the most important factor, we would need to study every single factor and compare each other to the number of callbacks.